src/QtUtils/MainThreadExecutorImplTest.cpp

```diff
@@ -266,4 +260,99 @@ TEST(MainThreadExecutorImpl, ScheduleAfterIfSuccessTwice) {
   EXPECT_EQ(second_chained_future.Get().value(), "The number is 42");
 }
 
-}  // namespace orbit_qt_utils
+TEST(MainThreadExecutorImpl, WaitForNoTimeout) {
+  auto executor = MainThreadExecutorImpl::Create();
+  orbit_base::Future<void> future = executor->Schedule([]() {});
+  EXPECT_EQ(executor->WaitFor(future), MainThreadExecutor::WaitResult::kCompleted);
+}
+
+TEST(MainThreadExecutorImpl, WaitForWithTimeout) {
+  auto executor = MainThreadExecutorImpl::Create();
+  orbit_base::Future<void> future = executor->Schedule([]() {});
+  EXPECT_EQ(executor->WaitFor(future, std::chrono::milliseconds{1000}),
+            MainThreadExecutor::WaitResult::kCompleted);
+}
+
+TEST(MainThreadExecutorImpl, WaitForNoTimeoutCompletedFutures) {
+  auto executor = MainThreadExecutorImpl::Create();
+  orbit_base::Future<void> future{};  // Constructs completed future
+  EXPECT_EQ(executor->WaitFor(future), MainThreadExecutor::WaitResult::kCompleted);
+}
+
+TEST(MainThreadExecutorImpl, WaitForWithTimeoutCompletedFutures) {
+  auto executor = MainThreadExecutorImpl::Create();
+  orbit_base::Future<void> future{};  // Constructs completed future
+  EXPECT_EQ(executor->WaitFor(future, std::chrono::milliseconds{1000}),
```

Comment:
does it work with 10? if not maybe change timeout in the next test to 1 second as well.

Reply:
I just tried that. Yes it works. (Ran the tests 10,000 times).

But I'm not sure what to gain from this change. Shouldn't make a difference functionality-wise and might potentially increase likelyhood of flaky test. What am I missing?

Reply:
the point here if this one does not work for some timeout the next test might yield false positive result. If a test times out for 10 on completed future we shouldn't assume it will not timeout on incorrectly working incomplete future.

Reply:
Got it. Okay. Makes sense. I will leave it at 10ms.